lib/runners/qemu.nix

```diff
@@ -339,16 +339,18 @@ in {
   setBalloonScript =
     if socket != null
     then ''
-      VALUE=$(( $SIZE * 1024 * 1024 ))
+      VALUE=$(( (${toString (mem + balloonMem)} - $SIZE) * 1024 * 1024 ))
       SIZE=$( (
         ${writeQmp { execute = "qmp_capabilities"; }}
-        ${writeQmp { execute = "balloon"; arguments.value = 987; }}
-      ) | sed -e s/987/$VALUE/ | \
+        (${writeQmp { execute = "balloon"; arguments.value = 987; }}) | sed -e s/987/$VALUE/
+        ${pkgs.coreutils}/bin/sleep 2
```

Comment:
The other VMMs also exhibit delayed effects. I think we should add the `sleep` to the other runners, too.

Also, the hardcoded `2` might be too low for loaded systems. Do you mind creating an option for it?

Reply:
Cloud-hypervisor didn't seem to require a delay, added delay to crosvm and made the delay configurable (still defaults to 2)